src/nectarchain/makers/component/tests/test_flatfield_spe_component.py

```diff
@@ -0,0 +1,287 @@
+from unittest.mock import patch
+
+import numpy as np
+import pytest
+from ctapipe.utils import get_dataset_path
+
+from nectarchain.data.container import SPEfitContainer
+from nectarchain.makers.component import (
+    ChargesComponent,
+    FlatFieldCombinedSPEStdNectarCAMComponent,
+    FlatFieldSingleHHVSPENectarCAMComponent,
+    FlatFieldSingleHHVSPEStdNectarCAMComponent,
+    FlatFieldSingleNominalSPENectarCAMComponent,
+    FlatFieldSingleNominalSPEStdNectarCAMComponent,
+)
+from nectarchain.makers.component.spe.spe_algorithm import SPEalgorithm
+from nectarchain.makers.core import BaseNectarCAMCalibrationTool
+
+
+class TestFlatFieldSingleNominalSPENectarCAMComponent:
+    RUN_NUMBER = 3938
+    RUN_FILE = get_dataset_path("NectarCAM.Run3938.30events.fits.fz")
+    NPIXELS = 1834
+    METHOD = "GlobalPeakWindowSum"
+    EXTRACTOR_KWARGS = {"window_width": 3, "window_shift": 1}
+    MULTIPROC = True
+    NPROC = 2
+    CHUNKSIZE = 1
+    ASKED_PIXELS_ID = [50, 100, 200, 300]
+    CLASS = FlatFieldSingleNominalSPENectarCAMComponent
+
+    @pytest.fixture
+    def eventsource(self):
+        return BaseNectarCAMCalibrationTool.load_run(
+            run_number=self.RUN_NUMBER, run_file=self.RUN_FILE
+        )
+
+    @pytest.fixture
+    def event(self, eventsource):
+        return next(eventsource.__iter__())
+
+    @pytest.fixture
+    def instance(self, eventsource):
+        parent = BaseNectarCAMCalibrationTool()
+        parent._event_source = eventsource
+        parent.run_number = self.RUN_NUMBER
+        parent.npixels = self.NPIXELS
+        return self.CLASS(
+            subarray=eventsource.subarray,
+            parent=parent,
+            extractor_kwargs=self.EXTRACTOR_KWARGS,
+            method=self.METHOD,
+            multiproc=self.MULTIPROC,
+            chunksize=self.CHUNKSIZE,
+            nproc=self.NPROC,
+            asked_pixels_id=self.ASKED_PIXELS_ID,
+        )
+
+    def test_init(self, instance):
+        assert isinstance(instance, self.CLASS)
+        assert isinstance(instance._SPEfitalgorithm_kwargs, dict)
+        assert instance._SPEfitalgorithm_kwargs == {
+            "multiproc": self.MULTIPROC,
+            "chunksize": self.CHUNKSIZE,
+            "nproc": self.NPROC,
+        }
+        assert isinstance(instance.chargesComponent, ChargesComponent)
+        assert instance._chargesContainers is None
+
+    def test_call(self, instance, event):
+        instance(event)
+        assert len(instance.chargesComponent.trigger_list) == 1
+
+    @pytest.mark.skip(reason="test multiproc make GitHub worker be killed")
```

Comment:
So we indeed need to avoid using multi-threading in GitHub workers, you confirm it, @guillaumegrolleron ?

Reply:
I think it's just a mistake, I forgot to reactivate these tests

Reply:
Ok these tests work well